### ISSUE TITLE
Compatibility with Email Templates admin editor add on

### ISSUE THREAD

**Issue by kimcoleman** [strangerstudios/pmpro-recurring-emails#8]
Can the email sent from this add on be compatible for editing in the email template editor add on?

**Comment by positivejam** [comment]
Yep, if the email template editor add on is installed, an email template for this add on is made available for editing on the Email Templates page in the backend.

**Comment by positivejam** [comment]
![image](https://user-images.githubusercontent.com/11378291/32138744-ce97492e-bbf5-11e7-99e4-2fa55c4c70a8.png)

**Comment by LMNTL** [comment]
While the email template does show up, the email sends out without some of the required variables. This is because the `$pmproemail->data` array only gets populated during `pmpror_recurring_emails`.

![image](https://user-images.githubusercontent.com/7819986/61570258-0be2da00-aa51-11e9-94e0-62a755635b02.png)


I've created this gist as a workaround: https://gist.github.com/LMNTL/fc22a6fc3d8079055f6f471475fd4fbc

But for a less hack-y solution we would need to refactor part of `pmpror_recurring_emails` to be its own function then use that function to filter `pmpro_email_data` (or something like that): https://github.com/strangerstudios/pmpro-recurring-emails/blob/dev/pmpro-recurring-emails.php#L162-L228

Forum ticket (mods only):
https://www.paidmembershipspro.com/forums/topic/bug-recurring-payment-email-reminders-add-on-email-variables-are-undefined/

**Comment by ideadude** [comment]
Related: https://github.com/strangerstudios/paid-memberships-pro/issues/624

This could be fixed in the core plugin, the email templates plugin, or the recurring emails plugin. I think we should fix it in the recurring emails plugin. The recurring emails plugin should define the data it expects to be used in the email itself.

What would be nice is if the pmproet_templates filter also expected a $templates['...']['data'] value and used that to populate the available variables when editing the email on the Memberships -> Email Templates page.

There is another gist similar to Jess T's floating around that just adds all the variables for all of the emails. This is a decent work around.

**Comment by tictag** [comment]
Hi, has this been fixed yet? I'm using Paid Memberships Pro - Recurring Emails Add On v0.5.4

Thanks.

**Comment by tictag** [comment]
Do you have a record for how long an issue has stayed an issue without resolution by Stranger Studios or the community? Is this one in the running? If not, which issue is the record holder?

**Comment by tictag** [comment]
Any news?

**Comment by trackwell-mike** [comment]
Same issue as above.
